frontend/src/metabase/admin/databases/containers/DatabaseEditApp.unit.spec.tsx

```diff
@@ -50,23 +67,25 @@ async function setup({ cachingEnabled = false, databaseIdParam = "" } = {}) {
     "enable-query-caching": cachingEnabled,
   });
 
-  renderWithProviders(
+  const { history } = renderWithProviders(
     <Route path="/">
+      <Route path="/home" component={TestHome} />
       <IndexRoute component={DatabaseEditApp} />
       <Route path=":databaseId" component={DatabaseEditApp} />
     </Route>,
     {
       withRouter: true,
-      initialRoute: `/${databaseIdParam}`,
+      initialRoute,
       storeInitialState: {
         settings,
       },
     },
   );
 
-  await waitForElementToBeRemoved(() => screen.queryByText("Loading..."));
```

Comment:
I tried to move this line to each test that needed it, but it failed in every single one (I put it right after `await setup()`).
Turns out that when `setup()` promise ends, the loader is already removed.

Interesting.

Reply:
I've had this discussion with @oisincoveney one time. He discussed with @ranquild that our unit tests aren't deterministic sometimes we can see the loader and sometimes we don't. `waitForElementToBeRemoved` requires the loader to always be present before it's removed.

So I think they came up with a solution that only uses `waitFor` e.g. (the referenced code isn't from the said discussion, I couldn't remember what PR that was)

https://github.com/metabase/metabase/blob/515f3da8fda3c3e51bf1b90402b35de59e7cfa07/frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.unit.spec.tsx#L258-L265

Maybe this will work for this case?

Reply:
^ yeah this is the one using `waitFor`. context is here https://metaboat.slack.com/archives/C505ZNNH4/p1684753502335459?thread_ts=1684751522.480859&cid=C505ZNNH4

Reply:
Nice, good workaround. TIL.
Thank you guys!

I updated all similar places that I recently have been dealing with: c05c1ba0f3869069072fafd4900a624bb798d45a